src/main/java/org/mariadb/jdbc/MariaDbClientPreparedStatement.java

```diff
@@ -189,19 +214,91 @@ public void addBatch(final String sql) throws SQLException {
         throw new SQLException("Cannot do addBatch(String) on preparedStatement");
     }
 
-    private void checkBatchFields() {
-        if (batchQueries == null) {
-            batchQueries = new ArrayList<>();
-        }
-    }
-
     /**
      * Clear batch.
      */
     @Override
     public void clearBatch() {
-        if (batchQueries != null) {
-            batchQueries.clear();
+        parameterList.clear();
+        this.parameters = new ParameterHolder[paramCount];
+    }
+
+    /**
+     * {inheritdoc}.
+     */
+    public int[] executeBatch() throws SQLException {
+        checkClose();
+        int size = parameterList.size();
+        if (size == 0) {
+            return new int[0];
+        }
+
+        int[] ret = new int[size];
+        int batchQueriesCount = 0;
+        MariaDbResultSet rs = null;
+        cachedResultSets.clear();
```

Comment:
cachedResultSets is an ArrayDeque, with the use of a lock below it makes me wonder if this structure needs to be synchronized/protected for thread safety?